### ISSUE TITLE
javac does not generate debug info by default

### ISSUE THREAD

**Issue by electricmonk** [davidB/scala-maven-plugin#116]
In maven-java-compiler, by default the compiler generates all appropriate debug info (-g:vars, -g:lines, -g:source). scala-maven-compiler does not do that, resulting in confusing runtime failures.

For instance, we're using Spring Framework @Configuration classes, which use method parameter names (from the Local Variable Table) to resolve dependency names. After switching a project (with mixed Java/Scala code) to scala-maven-plugin, Spring dependencies broke and it took me some time to figure out the cause.

In short, please pass -g to javac by default, as otherwise stuff can get screwed up and people won't necessarily realize why.
